src/sandbox.ml

```diff
@@ -282,13 +296,22 @@ let detect_opam_sandbox ~project_root opam () =
   Opam (opam, switch)
 ;;
 
+let detect_dune_lock_dir ~project_root () =
+  let open Promise.Syntax in
+  (* Path to the dune.lock file *)
+  let dune_lock_path = Path.join project_root (Path.of_string "dune.lock") in
+  let+ exists = Fs.exists (Path.to_string dune_lock_path) in
+  if exists then Some (Dune "$prog $args") else None
```

Comment:
Again, you never use the information that you store in the `Dune` constructor: `"$prog $args"` you should put nothing, or better, store the project root.